hubblestack/files/hubblestack_nova/oval_scanner.py

```diff
@@ -0,0 +1,428 @@
+"""
+This is a Hubble Nova CVE scanner that uses an OVAL source file.
+
+Written by Wes Miser
+Contributions by Michael Robinson and Proofpoint, Inc.
+
+To use this scanner, an oval.yaml file must exist within the
+hubblestack_nova_profiles directory, preferably within the cve folder.  The
+contents of the file should be as follows (notice everything under oval_scanner
+is indented):
+
+oval_scanner:
+  opt_baseurl: <valid http/https url>
+  opt_remote_sourcefile: <valid source file>
+  opt_local_sourcefile: <valid source file residing on the client>
+  opt_output_file: <write the vulnerability results to a file on the client>
+
+oval_scanner is the primary key.  Each opt key's value under the primary key
+is optional and does not need to be specified.  If opt_baseurl,
+opt_remote_sourcefile, and opt_local_sourcefile values are not specified, the
+scanner will automatically pull the appropriate OVAL source definition file from
+the supported distro's public repository.  opt_local_sourcefile will override
+opt_baseurl and opt_remote_sourcefile even if their values are specified.
+
+top.nova must also reference the oval.yaml file.  Note that other CVE scanners
+must be disabled or conflicts will ensue.  The contents of top.nova can look as
+follows:
+
+nova:
+  'G@kernel:Linux and not G@osfinger:*CoreOS*':
+    ## - cve.vulners <-- ensure other CVE scanners are not active
+    - cve.oval
+
+When run, the scanner will parse the source OVAL file into a readable
+dictionary, maps OVAL defintions directly to OVAL object and OVAL state
+references based on OVAL test reference data of the definition, and then makes
+a comparison to the local packages installed on the system to identify potential
+vulnerabilities.
+
+This scanner currently only supports the Linux platform.
+"""
+
+from __future__ import absolute_import
+
+import xml.etree.ElementTree as ET
+import json
+import sys
+import requests
+import logging
+
+
+def __virtual__():
+    return not sys.platform.startswith('win')
```

Comment:
Could you switch to using [`salt.utils.platform.is_windows()`](https://github.com/hubblestack/hubble/blob/9849c4cb342acf1dc08f5b4b4d57216a010caee8/hubblestack/files/hubblestack_nova/win_firewall.py#L19) just for consistency across the codebase? Obviously if we end up removing the salt import it will need to be changed, but it's easier of all of these checks look the same.

Reply:
I'll implement these changes.